node/impl/full/eth.go

```diff
@@ -890,6 +890,15 @@ func (a *EthModule) EthEstimateGas(ctx context.Context, tx ethtypes.EthCall) (et
 	ts := a.Chain.GetHeaviestTipSet()
 	msg, err = a.GasAPI.GasEstimateMessageGas(ctx, msg, nil, ts.Key())
 	if err != nil {
+		// On failure, GasEstimateMessageGas doesn't actually return the invocation result,
+		// it just returns an error. That means we can't get the revert reason.
+		//
+		// So we re-execute the message with EthCall (well, applyMessage which contains the
+		// guts of EthCall). This will give us an ethereum specific error with revert
+		// information.
+		if _, err2 := a.applyMessage(ctx, msg, ts.Key()); err2 != nil {
```

Comment:
Can we get a test for this?

Reply:
The main logic should already be covered by the test in the previous patch. Although I guess I can just extend that test to call this as well?

Reply:
Oh, well, of course. It panics. Because I didn't write a test.

Reply:
Always the simple patches that can't possibly be wrong.